js/main.js

```diff
@@ -40,24 +40,37 @@ function showView(targetView) {
   }
 }
 
-// Submit new entry
-
+// Submit or Edit, then Render Entry
 function logSubmit(event) {
   event.preventDefault();
-  var entry = {
-    title: $form.elements.title.value,
-    photoURL: $form.elements.photoURL.value,
-    notes: $form.elements.notes.value,
-    entryId: data.nextEntryId
-  };
-  data.nextEntryId++;
-  data.entries.unshift(entry);
-
+  if (data.editing === null) {
+    var entry = {
+      title: $form.elements.title.value,
+      photoURL: $form.elements.photoURL.value,
+      notes: $form.elements.notes.value,
+      entryId: data.nextEntryId
+    };
+
+    data.nextEntryId++;
+    data.entries.unshift(entry);
+    var newEntry = renderEntry(entry);
+    $ul.prepend(newEntry);
+
+  } else {
+    var edited = {
+      title: $title.value,
+      photoURL: $photoURL.value,
+      notes: $notes.value,
+      entryId: data.editing.entryId
+    };
+
+    data.entries[data.editing.entryId] = edited;
```

Comment:
this won't work once you start deleting entries because the array indexes won't correspond to the length in any meaningful way. make sure to search through the entries array and find the entry by the id!